types/socket.io/index.d.ts

```diff
@@ -787,6 +787,11 @@ declare namespace SocketIO {
          */
         headers: any;
 
+        /**
+         * The authentication payload
+         */
+        auth: { [props: string]: any };
```

Comment:
Is not the `token` always there? Please try to narrow it as much as possible.

Reply:
Oops! I thought it may has different props.